src/xo-app/settings/remotes/index.js

```diff
@@ -0,0 +1,313 @@
+      <td><ActionButton btnStyle='danger' handler={deleteRemote} handlerParam={remote.id} icon='delete' /></td>
+    </tr>
+  }
+
+  _renderRemoteInfo (remote) {
+    throw new Error('NOT IMPLEMENTED')
+  }
+
+  _renderAuthInfo (remote) {
+    throw new Error('NOT IMPLEMENTED')
+  }
+
+  get accessible () {
+    throw new Error('NOT IMPLEMENTED')
+  }
+
+  get unaccessible () {
+    throw new Error('NOT IMPLEMENTED')
+  }
+}
+
+class LocalRemote extends AbstractRemote {
+  _renderRemoteInfo (remote) {
+    return remote.path
+  }
+
+  _renderAuthInfo (remote) {
+    return ''
+  }
+
+  get accessible () {
+    return 'Accessible'
+  }
+
+  get unaccessible () {
+    return 'Unaccessible'
+  }
+}
+
+class NfsRemote extends AbstractRemote {
+  _renderRemoteInfo (remote) {
+    return `${remote.host}:${remote.share}`
+  }
+
+  _renderAuthInfo (remote) {
+    return ''
+  }
+
+  get accessible () {
+    return 'Mounted'
+  }
+
+  get unaccessible () {
+    return 'Unmounted'
+  }
+}
+
+class SmbRemote extends AbstractRemote {
+  _renderRemoteInfo (remote) {
+    return <span><strong className='text-info'>\\</strong>{remote.host}<strong className='text-info'>\</strong>{remote.path}</span>
+  }
+
+  _renderAuthInfo (remote) {
+    return <span>{remote.username}@{remote.domain}</span>
+  }
+
+  get accessible () {
+    return 'Accessible'
+  }
+
+  get unaccessible () {
+    return 'Unaccessible'
+  }
+}
+
+@injectIntl
+@connectStore({})
+export default class New extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      remotes: {},
+      type: 'local'
+    }
+  }
+
+  componentWillMount () {
+    this.componentWillUnmount = subscribeRemotes(rawRemotes => {
+      rawRemotes = map(rawRemotes, parse)
+      const remotes = {}
+      for (let type in remoteTypes) {
+        remotes[type] = filter(rawRemotes, r => r.type === type)
+      }
+      this.setState({
+        remotes
+      })
+    })
+  }
+
+  _handleRemoteTypeSelection = type => this.setState({type})
+
+  _createRemote = async () => {
+    const {
+      name,
+      host,
+      path,
+      username,
+      password,
+      domain
+    } = this.refs
+    const {
+      type
+    } = this.state
+
+    const UI_TYPE_REMOTE_TYPE = {
+      local: 'file',
+      nfs: 'nfs',
+      smb: 'smb'
+    }
+
+    const urlParams = {
+      type: UI_TYPE_REMOTE_TYPE[type],
+      host: host && host.value,
+      path: path && path.value
+    }
+    username && (urlParams.username = username.value)
+    password && (urlParams.password = password.value)
+    domain && (urlParams.domain = domain.value)
+
+    const url = format(urlParams)
+    return createRemote(name && name.value, url)
+    .then(() => {
+      this.setState({type: 'local'})
+      path && (path.value = '')
+      username && (username.value = '')
+      password && (password.value = '')
+      domain && (domain.value = '')
+    }, err => error('Create Remote', err.message || String(err)))
```

Comment:
I'm wondering whether it would make sense to do error handling directly in `xo.js`.